service/history/timerSequence.go

```diff
@@ -101,8 +101,8 @@ func (t *timerSequenceImpl) isExpired(
 ) bool {
 
 	// Cassandra timestamp resolution is in millisecond
-	// here we do the check in terms of second resolution.
-	return timerSequenceID.timestamp.Unix() <= referenceTime.Unix()
+	println(fmt.Sprintf("timerSeq %v - referenceTime %v", timerSequenceID.timestamp, referenceTime))
+	return (timerSequenceID.timestamp.UnixNano() / 10e6) <= (referenceTime.UnixNano() / 10e6)
```

Comment:
@samarabbas 
https://github.com/temporalio/temporal/issues/245

Reply:
@wxing1292 this is a draft PR and not ready for review yet.  We are chasing an issue where timers fires earlier than expected due to granularity of timers.  Can you clarify how this could be related to #245?

Reply:
t = 0
a timer decision task trigger a generation of timer task, say timer task A

t = 1
a cancel timer decision cancel the above timer, in mutable state only

t = 2
a timer decision task trigger a generation of timer task, say timer task B
NOTE: 
abs(timer task A - timer task B) < precision
timer task A fires before timer task B

t = n
timer task A is triggered, although this task should be thrown away, due to precision issue, this task will still fire.

Reply:
btw, is the java client (timer state machine) still relying on the timestamp of events? if so, i would suggest do the same thing like the go client.